functions/auth.js

```diff
@@ -10,3 +10,12 @@ exports.checkAuth = (context) => {
     throw new functions.https.HttpsError("permission-denied", "Token invalid");
   }
 };
+
+// Case-insensitive first character for iDevices
+const lowercaseFirstChar = (password) =>
+  password.charAt(0).toLowerCase() + password.substring(1);
+
+exports.passwordsMatch = (submittedPassword, actualPassword) =>
+  submittedPassword.trim() === actualPassword.trim() ||
+  lowercaseFirstChar(submittedPassword.trim()) ===
+    lowercaseFirstChar(actualPassword.trim());
```

Comment:
This will reduce the entropy of the passwords.

What happens if the password validly starts/ends with spaces?

Reply:
It is not a common practice to allow spaces in password fields. Or at least I have not seen it anywhere. Maybe we should discuss this with Ed if we should allow users to do so or not.

Reply:
I'm not against us disallowing spaces at the start/end of the password field, but if that is our choice, that should be enforced through the form validation where the password is created/edited/not allowing an invalid password to be saved, not at the 'end of the chain' where we are comparing them.

Reply:
https://github.com/sparkletown/internal-sparkle-issues/issues/187